Asterix/optics/propagation_functions.py

```diff
@@ -610,30 +615,81 @@ def prop_fpm_regional_sampling(pup,
     EF_before_LS : 2D array (complex)
         E-field before the Lyot stop.
     """
-    from Asterix.optics import shift_phase_ramp
+    samp_outer = 2
+
+    if not isinstance(nbres, (list, np.ndarray)):
```

Comment:
So are arrays allowed for `nbres` or not? You deleted the array as an option from the docstring, so I am confused.

Reply:
corrected